internal/attestation/azure/snp/validator.go

```diff
@@ -36,249 +41,368 @@ type Validator struct {
 	if err != nil {
-		return nil, fmt.Errorf("validating VCEK: %w", err)
-	}
-
-	if err := v.validateSNPReport(ctx, vcek, report, instanceInfo.MAAToken, extraData); err != nil {
-		return nil, fmt.Errorf("validating SNP report: %w", err)
-	}
-
+		return nil, fmt.Errorf("parsing ASK certificate: %w", err)
+	}
+	// We verify the ASK certificate against the ARK certificate, since the ARK certificate is the root certificate
+	// and inherently trusted. This check should also be performed by go-sev-guest, but as we have to specify both ASK and ARK for
+	// the trusted roots, this needs to be done here as well.
+	rootPool := x509.NewCertPool()
+	rootPool.AddCert(&trustedArk)
+	if _, err := ask.Verify(x509.VerifyOptions{
+		Roots: rootPool,
+	}); err != nil {
+		return nil, fmt.Errorf("verifying ASK certificate: %w", err)
+	}
+	if err := v.attestationVerifier.SnpAttestation(att, &verify.Options{
+		TrustedRoots: map[string][]*trust.AMDRootCerts{
+			"Milan": {
+				{
+					Product: "Milan",
+					ProductCerts: &trust.ProductCerts{
+						Ask: ask,
+						Ark: &trustedArk,
+					},
+				},
+			},
+		},
+	}); err != nil {
+		return nil, fmt.Errorf("verifying SNP attestation: %w", err)
+	}
+
+	// Checks if the attestation report matches the given constraints.
+	// Some constraints are implicitly checked by validate.SnpAttestation:
+	// - the report is not expired
+	if err := v.attestationValidator.SnpAttestation(att, &validate.Options{
+		GuestPolicy: abi.SnpPolicy{
+			Debug: false, // Debug means the VM can be decrypted by the host for debugging purposes and thus is not allowed.
+			SMT:   true,  // Allow Simultaneous Multi-Threading (SMT). Normally, we would want to disable SMT
+			// but Azure does not allow to disable it.
+		},
+		VMPL: new(int), // Checks that Virtual Machine Privilege Level (VMPL) is 0.
+		// This checks that the reported TCB version is equal or greater than the minimum specified in the config.
+		MinimumTCB: kds.TCBParts{
+			BlSpl:    v.config.BootloaderVersion.Value, // Bootloader
+			TeeSpl:   v.config.TEEVersion.Value,        // TEE (Secure OS)
+			SnpSpl:   v.config.SNPVersion.Value,        // SNP
+			UcodeSpl: v.config.MicrocodeVersion.Value,  // Microcode
+		},
+		// This checks that the reported LaunchTCB version is equal or greater than the minimum specified in the config.
+		MinimumLaunchTCB: kds.TCBParts{
+			BlSpl:    v.config.BootloaderVersion.Value, // Bootloader
+			TeeSpl:   v.config.TEEVersion.Value,        // TEE (Secure OS)
+			SnpSpl:   v.config.SNPVersion.Value,        // SNP
+			UcodeSpl: v.config.MicrocodeVersion.Value,  // Microcode
+		},
+		// Check that CurrentTCB >= CommittedTCB.
+		PermitProvisionalFirmware: true,
+		// Check if the IDKey hashes in the report are in the list of accepted hashes.
+		TrustedIDKeyHashes: v.config.FirmwareSignerConfig.AcceptedKeyDigests,
+		// The IDKey hashes should not be checked if the enforcement policy is set to MAAFallback or WarnOnly to prevent
```

Comment:
```suggestion
		// Check if the IDKey hash in the report is in the list of accepted hashes.
		TrustedIDKeyHashes: v.config.FirmwareSignerConfig.AcceptedKeyDigests,
		// The IDKey hash should not be checked if the enforcement policy is set to MAAFallback or WarnOnly to prevent
```